modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -255,12 +255,14 @@ def get_files_info(ip, shares)
             sz = file.end_of_file
           end
 
+          # Logging of the obtained data.
+          logdata << "#{ip}\\#{share_name}#{subdirs.first}\\#{fname.encode}\n"
+
           # Filename is too long for the UI table, cut it.
           fname = "#{fname[0, 35]}..." if fname.length > 35
 
           pretty_tbl << [fa || 'Unknown', fname, tcr, tac, twr, tch, sz]
           detailed_tbl << [ip.to_s, fa || 'Unknown', share_name, subdirs.first + '\\', fname, tcr, tac, twr, tch, sz]
-          logdata << "#{ip}\\#{share_name}#{subdirs.first}\\#{fname.encode}\n"
```

Comment:
This isn't a bad solution but I see one other problem in that detailed_tbl is also saved to loot on line 273/275 of this output and at line 276/278. Therefore we would still have an issue with those lines of code logging a truncated version of `fname` to the logs.

Just something I noticed in addition to Alan's original comment when reviewing this.

We can safely ignore `pretty_tbl` since that is only printed out and is not saved to loot using `store_loot`.

Reply:
Understood, thanks. Pushing the commits with the suggested changes.

Reply:
Resolved with https://github.com/rapid7/metasploit-framework/pull/17454/commits/2df39204325aedbd54ad34a73ae2d43cb51af6a0